openstef/model/regressors/arima.py

```diff
@@ -0,0 +1,163 @@
+# SPDX-FileCopyrightText: 2017-2022 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""This module contains the SARIMAX regressor wrapper around statsmodels implementation."""
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+
+from sklearn.metrics import r2_score
+from sklearn.model_selection import TimeSeriesSplit
+from openstef.model.regressors.regressor import OpenstfRegressor
+
+
+class ARIMAOpenstfRegressor(OpenstfRegressor):
+    """Wrapper around statmodels implementation of (S)ARIMA(X) model.
+
+    To make prediction, it needs to update its historic, applying the fitted parameters to new data unrelated to the
+    original training data.
+
+    """
+
+    def __init__(
+        self,
+        backtest_max_horizon=1440,
+        order=(0, 0, 0),
+        seasonal_order=(0, 0, 0, 0),
+        trend=None,
+    ):
+        self.backtest_max_horizon = backtest_max_horizon
+        self.order = order
+        self.seasonal_order = seasonal_order
+        self.trend = trend
+
+    def fit(self, x, y, **kwargs):
+        exog = x
```

Comment:
Maybe add a link to https://www.statsmodels.org/stable/endog_exog.html to give a bit more background on the exog terminology as it is not that common.

Reply:
Great suggestion: 
- I added multiple links to relevant documentations of statsmodels
- I removed the renaming of variables and explicitly passed the arguments as keyword argument to be clearer.